examples/eSealing/src/lib.rs

```diff
@@ -0,0 +1,444 @@
+//! A contract implementing an eSealing service.
+//!
+//! Only accounts can register a file hash. During the registration, the
+//! timestamp is recorded together with the witness (sender_account) that
+//! registered the file hash.
+//!
+//! Every account can only register each file hash
+//! once. This is because we want to record the time when the witness
+//! (sender_account) first had access to that file. Re-registering a file hash
+//! by the same account would not give us any additional information.
+//!
+//! But: A second account can register a file hash that was already previously
+//! registered. This second account will be added as another witness together
+//! with the timestamp when the second account called the `register_file`
+//! function. We know that the second account is most likely now also in
+//! possession of the file if it has used this eSealing service via the
+//! front-end. But this is not guaranteed since the second account could
+//! register a random file hash it found in the events by interacting with the
+//! smart contract directly (without using the front-end).
```

Comment:
Seeing that you already point out the issue with the second registration. What is the use-case for it?

We cannot use it for a guarantee. So I'm not sure why we have the added complexity.